rook/trees/TreeStructure.py

```diff
@@ -0,0 +1,1327 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on Jan 28, 2014
+@ author: alfoa
```

Comment:
If this is an exact copy of the utilities in RAVEN, can we note that somewhere?

Reply:
Adding comment about that in the docstring.